code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -170,29 +168,43 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	if(smoketime < 1)
 		die()
 		return
+	smoke()
+	return
+
+
+/obj/item/clothing/mask/cigarette/attack_self(mob/user as mob)
+	if(lit)
+		user.visible_message("<span class='notice'>[user] calmly drops and treads on the lit [src], putting it out instantly.</span>")
+		die()
+	return ..()
+
+/obj/item/clothing/mask/cigarette/proc/smoke()
+	var/turf/location = get_turf(src)
+	var/is_being_smoked = 0
+	// Check whether this is actually in a mouth, being smoked
+	if(iscarbon(loc))
+		var/mob/living/carbon/C = loc
+		if(src == C.wear_mask)
+			if(istype(loc, /mob/living/carbon/human))
+				var/mob/living/carbon/human/H = loc
+				if(!(H.species.flags & IS_SYNTHETIC)) // Silly synthetics, you can't smoke. You don't even have mouths.
```

Comment:
This doesn't actually affect whether they can have a lit cigarette smooshed against their monitor, only whether it's transferring reagents to them. Also, smokables not transferring reagents to IPCs is the existing behavior - I simply reorganized the code, and fixed the "smokables in an IPC's mouth don't burn reagents" bug (which was a result of sloppy coding).

Considering IPCs process reagents now, should I change it so they get reagents from smokables? The fact that they still don't may simply be an oversight.